src/core/backend/pearl/core.rs

```diff
@@ -73,6 +75,40 @@ impl Pearl {
                 Error::failed(format!("cannot find actual alien folder. {:?}", operation))
             })
     }
+
+    async fn count_blobs(groups: Arc<Vec<Group>>, alien_groups_guard: Arc<RwLock<Vec<Group>>>,
```

Comment:
Please, recheck that the references (`group` and `alien_groups_guard`) are never changed during runtime. I.e. some other code does not perform collection reallocation.

Reply:
Yes, I've rechecked that and discussed with @pyakushin . There is no problem because groups doesn't change during runtime of the program (once they are built; you can check it: no method in public interface of Pearl (BackendStorage) that changes groups vector) and alien groups are under RWLock, so they can't be changed while I iterate on them.